Rakefile

```diff
@@ -71,6 +75,63 @@ def re_run_basic_smoke_test
   sh beaker
 end
 
+def get_passing_json(current_branch_trunk)
+  jenkins_url = "https://jenkins-master-prod-1.delivery.puppetlabs.net/view/puppet-agent/view/#{current_branch_trunk}/view/Suite/job/platform_puppet-agent_intn-van-promote_suite-daily-promotion-#{current_branch_trunk}/lastSuccessfulBuild/api/json"
+  @memoized_passing_json ||= Net::HTTP.get(URI.parse(jenkins_url))
```

Comment:
I would avoid memoizing a value that is dependent on the parameter passed in, unless you check that the value passed in is the same as the one used to compute the memoized value.

It might be nice to hit Jenkins in the rake task and then pass the JSON, or deserialized hash, into the methods that would otherwise call this also at within the top of the rake task, then passing their values into the replace_pins method, removing the need to memoize in the first place.